engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/blueprint/NadelExecutionBlueprintFactory.kt

```diff
@@ -205,19 +212,88 @@ private class Factory(
         val location = makeFieldCoordinates(parentType, hydratedFieldDef)
 
         val batchSize = hydration.batchSize ?: 50
+        val hydrationArgs = getHydrationArguments(hydration, parentType, hydratedFieldDef, actorFieldDef)
+
+        val matchStrategy = if (hydration.isObjectMatchByIndex) {
+            NadelBatchHydrationMatchStrategy.MatchIndex
+        } else if (hydration.objectIdentifier.contains(",")) {
```

Comment:
I _will_ change this to the complex object thing like below

```graphql
identifiedBy: [
      {source: "complexIssueId.id", target: "id"}
      {source: "complexIssueId.site", target: "site"}
]
```

Just made the test work for now